CHANGELOG.md

```diff
@@ -1,5 +1,93 @@
 # Changelog
 
+## [v0.4.0](https://github.com/delta-incubator/delta-kernel-rs/tree/v0.4.0/) (2024-10-22)
+
+[Full Changelog](https://github.com/delta-incubator/delta-kernel-rs/compare/v0.3.1...v0.4.0)
+
+**API Changes**
+
+*Breaking*
+
+1. `pub ScanResult.mask` field made private and only accessible as `ScanResult.raw_mask()` method [\#374]
+2. new `ReaderFeatures` enum variant: `TypeWidening` and `TypeWideningPreview` [\#335]
+3. new `WriterFeatures` enum variant: `TypeWidening` and `TypeWideningPreview` [\#335]
+4. new `Error` enum variant: `InvalidLogPath` when kernel is unable to parse the name of a log path [\#347]
+5. Module moved: `mod delta_kernel::transaction` -> `mod delta_kernel::actions::set_transaction` [\#386]
+6. output of arrow expression evaluation now applies/validates output schema in default arrow expression handler [\#331]
```

Comment:
Hm yea I had the same question (where we should put those changes to expression evaluation). I agree it "allows previously broken stuff to no longer break" and since the semantics remain the same (just that we are enforcing/implementing it now) perhaps this goes under 'fixes'? WDYT @nicklan @scovich

Reply:
Yeah, fixes is probably good. I don't know how critical it is that we get the changelog "just right" at this point, but nothing will break due to this change (i don't think) so fixes or enhancements makes sense.